app/notification_push.go

```diff
@@ -132,8 +140,20 @@ func (a *App) sendPushNotification(notification *postNotification, user *model.U
 	channelName := notification.GetChannelName(nameFormat, user.Id)
 	senderName := notification.GetSenderName(nameFormat, *cfg.ServiceSettings.EnablePostUsernameOverride)
 
+	notificationId := model.NewId()
```

Comment:
you are already receiving this id as a parameter, correct?

Reply:
No here, this is the `sendPushNotification` which is the async part which register the notification to be sent.